### DISCUSSION TITLE
Help using i2c instead of SPI

### DISCUSSION THREAD

**Discussion by Lavertronix** [tttapa/Control-Surface#780] [Q&A]
I know this question has come up before but I am still not sure how to use an i2c display instead of SPI in the vu meter sketch. I am trying to use the Adafruit library but have not had any luck. Can someone maybe post a modified sketch or explain exactly what I need to do, please?

**Comment by tttapa** [comment]
Replace the initialization here

https://github.com/tttapa/Control-Surface/blob/ebec32b97fd474ba21cb1e31b7a06ea7200233c1/examples/Boards/ESP32/OLED/VU-Meter-Bridge/VU-Meter-Bridge.ino#L80-L82

by something more along the lines of 

https://github.com/adafruit/Adafruit_SSD1306/blob/88711efc3e4b4a1a3fb238a1c5aa9590697f9c02/examples/ssd1306_128x64_i2c/ssd1306_128x64_i2c.ino#L36

and add the I²C address to the call to `begin()` here

https://github.com/tttapa/Control-Surface/blob/ebec32b97fd474ba21cb1e31b7a06ea7200233c1/examples/Boards/ESP32/OLED/VU-Meter-Bridge/VU-Meter-Bridge.ino#L95

as shown in 

https://github.com/adafruit/Adafruit_SSD1306/blob/88711efc3e4b4a1a3fb238a1c5aa9590697f9c02/examples/ssd1306_128x64_i2c/ssd1306_128x64_i2c.ino#L64
